spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/ColumnarBatchReader.java

```diff
@@ -78,63 +76,13 @@ public final ColumnarBatch read(ColumnarBatch reuse, int numRowsToRead) {
           "Number of rows in the vector %s didn't match expected %s ", numRowsInVector,
           numRowsToRead);
 
-      if (rowIdMapping == null) {
-        arrowColumnVectors[i] = IcebergArrowColumnVector.forHolder(vectorHolders[i], numRowsInVector);
-      } else {
-        int[] rowIdMap = rowIdMapping.first();
-        Integer numRows = rowIdMapping.second();
-        arrowColumnVectors[i] = ColumnVectorWithFilter.forHolder(vectorHolders[i], rowIdMap, numRows);
-      }
+      arrowColumnVectors[i] = batch.hasDeletes() ?
```

Comment:
I think I would pass the vectors directly into the BatchWithFilte so everything like this check can be done inside that class as well. Then this class just creates the the BatchWith Filter with the proper vectors and then call .batch which would return a final columnar Batch from inside the class?

Reply:
I experimented that way. It turned out almost nothing left in the class `ColumnBatchReader`. So I asked myself why I should have another class.

Reply:
From what I can tell, our main reason we want to divide this is that ColumnarBatchReader can't build this entire object in its constructor. So It's probably fine to keep this as an inner class of ColumnarBatchReader.  I just want to keep as much state in a final  constructor as possible. Maybe that's a bit weird

Reply:
I mean ... I think even the reader's could be passed as an arg. Then the only things that Columnar Batch Reader is responsible for is for getting the readers and start position?

Reply:
Changed it to an inner class and moved vector read logic into the inner class.